hummingbot/connector/exchange/ndax/ndax_api_order_book_data_source.py

```diff
@@ -222,40 +227,37 @@ async def listen_for_order_book_diffs(self, ev_loop: asyncio.BaseEventLoop, outp
-
-                    msg_data: List[NdaxOrderBookEntry] = [NdaxOrderBookEntry(*entry)
-                                                          for entry in payload]
-                    msg_timestamp: int = max([e.actionDateTime for e in msg_data])
-                    msg_product_code: int = msg_data[0].productPairCode
-
-                    content = {"data": msg_data}
-                    msg_trading_pair: Optional[str] = None
-
-                    for trading_pair, instrument_id in self._trading_pair_id_map.items():
-                        if msg_product_code == instrument_id:
-                            msg_trading_pair = trading_pair
-                            break
-
-                    if not msg_trading_pair:
-                        continue
-
-                    metadata = {
-                        "trading_pair": msg_trading_pair,
-                        "instrument_id": msg_product_code,
-                    }
-
-                    if msg_event == CONSTANTS.WS_ORDER_BOOK_CHANNEL:
-                        snapshot_msg: OrderBookMessage = NdaxOrderBook.snapshot_message_from_exchange(msg=content,
+                    if msg_event in [CONSTANTS.WS_ORDER_BOOK_CHANNEL, CONSTANTS.WS_ORDER_BOOK_L2_UPDATE_EVENT]:
+                        msg_data: List[NdaxOrderBookEntry] = [NdaxOrderBookEntry(*entry)
+                                                              for entry in payload]
+                        msg_timestamp: int = max([e.actionDateTime for e in msg_data])
+                        msg_product_code: int = msg_data[0].productPairCode
+
+                        content = {"data": msg_data}
+                        msg_trading_pair: Optional[str] = None
+
+                        for trading_pair, instrument_id in self._trading_pair_id_map.items():
+                            if msg_product_code == instrument_id:
+                                msg_trading_pair = trading_pair
+                                break
+
+                        if msg_trading_pair:
+
+                            metadata = {
+                                "trading_pair": msg_trading_pair,
+                                "instrument_id": msg_product_code,
+                            }
+
+                            if msg_event == CONSTANTS.WS_ORDER_BOOK_CHANNEL:
+                                snapshot_msg: OrderBookMessage = NdaxOrderBook.snapshot_message_from_exchange(msg=content,
+                                                                                                              timestamp=msg_timestamp,
+                                                                                                              metadata=metadata)
+                                output.put_nowait(snapshot_msg)
+                            elif msg_event == CONSTANTS.WS_ORDER_BOOK_L2_UPDATE_EVENT:
+                                diff_msg: OrderBookMessage = NdaxOrderBook.diff_message_from_exchange(msg=content,
```

Comment:
The reason why I decided to include the clause to `continue` when (1) receiving an unrecognised msg event and (2) receiving updates for other trading pairs is to make it st it wont be an indentation hell like this. But nonetheless I feel both achieve the same thing,

Reply:
I agree with you regarding the indentation. But I think it would be better to solve that problem by extracting code to methods instead of using the continue statement.